Description: tune flags to build libv8 on armel
 * mno-thumb-interwork to allow compilation on default debian armel target (-march=armv4t)
 * fno-tree-sink to workaround a gcc armel bug
 * Wno-psabi to fix gcc warnings on armel
Forwarded: not-needed
Author: Antonio Radici <antonio@dyne.org>
Author: Jérémy Lal <kapouer@melix.org>
Bug-V8: http://code.google.com/p/v8/issues/detail?id=590
Bug-GCC: http://gcc.gnu.org/bugzilla/show_bug.cgi?id=39604
Bug-GCC: http://gcc.gnu.org/bugzilla/show_bug.cgi?id=42748
Last-Update: 2010-11-25
--- a/SConstruct
+++ b/SConstruct
@@ -140,6 +140,7 @@
     },
     'arch:arm': {
       'CPPDEFINES':   ['V8_TARGET_ARCH_ARM'],
+      'CCFLAGS':      ['-Wno-psabi', '-fno-tree-sink'],
       'unalignedaccesses:on' : {
         'CPPDEFINES' : ['CAN_USE_UNALIGNED_ACCESSES=1']
       },
--- a/src/arm/macro-assembler-arm.cc
+++ b/src/arm/macro-assembler-arm.cc
@@ -60,9 +60,6 @@
 // We do not support thumb inter-working with an arm architecture not supporting
 // the blx instruction (below v5t).  If you know what CPU you are compiling for
 // you can use -march=armv7 or similar.
-#if defined(USE_THUMB_INTERWORK) && !defined(CAN_USE_THUMB_INSTRUCTIONS)
-# error "For thumb inter-working we require an architecture which supports blx"
-#endif
 
 
 // Using bx does not yield better code, so use it only when required
